x-pack/plugins/security_solution/server/lib/detection_engine/rule_types/ml/ml.ts

```diff
@@ -120,6 +136,7 @@ export const mlExecutor = async ({
       return result;
     }
 
+    // TODO we add the max_signals warning _before_ filtering the anomalies against the exceptions list. Is that correct?
```

Comment:
@marshallmain @yctercero cleaning up my TODOs and wanted to call this out. Thoughts?

Reply:
To be completely accurate we would split the warning into 2 checks/warnings: the first one, before filtering the events against list exceptions, would check if the search result has more total hits than the page size and warn that some search results were missed. The second one would take the alerts after filtering events against list exceptions and check if the remaining number of alerts is greater than `maxSignals`.

Then we might want to make the search page size larger than max signals to make it more likely that we'll be able to create as many alerts as possible even if there are duplicates or some are filtered by list exceptions.